File: java/chaincue-real-estate/src/main/java/com/chaincuerealestate/chaincuerealestate/domains/House.java
package com.chaincuerealestate.chaincuerealestate.domains;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

import java.util.List;
import java.util.UUID;

@Entity(name = "house")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class House {
    @Id
    private String id;
    private String title;
    private String location;
    private int numberRooms;
    private int beds;
    private String price;
    @NonNull
    private String src;
    private boolean sold;
    @NonNull
    private String houseTypes;
    @OneToMany(cascade = CascadeType.ALL)
    private List<HouseImage> images;
    @OneToOne(cascade = CascadeType.ALL)
    private Broker broker;

    /*TODO add created and mock data*/
    public enum HouseTypes {
        CONDOMINIUM,
        VILLA,
        TOWNHOUSE,
        VACATION_HOME,
        ESTATES_AND_FARMS,
        LAND,
        OTHER_HOUSES
    }

    public static House create(HouseTypes houseTypes, String src) {
        return new House(
                UUID.randomUUID().toString(),
                "",
                "",
                0,
                0,
                "",
                src,
                false,
                houseTypes.toString(),
                List.of(),
                null
        );
    }

}

File: kotlin/chaincue-real-estate/src/main/kotlin/com/example/chaincuerealestate/domains/House.kt
package com.example.chaincuerealestate.domains

import jakarta.persistence.*
import lombok.NonNull
import java.time.LocalDateTime
import java.util.*

@Entity(name = "house")
data class House(
    @Id
    var id: String,
    var title: String,
    var description: String,
    var location: String,
    var numberRooms: Int,
    var beds: Int,
    var price: String,
    @get:NonNull
    var src: String,
    var sold: Boolean,
    @get:NonNull
    var houseTypes: String,
    @OneToMany(cascade = [CascadeType.ALL])
    var images: List<HouseImage>,
    @OneToOne(cascade = [CascadeType.ALL])
    var broker: Broker?,
    var timestamp: LocalDateTime
) {
    companion object {
        fun create(houseTypes: HouseTypes, src: String): House {
            return House(
                id = UUID.randomUUID().toString(),
                title = "",
                description = "",
                location = "",
                numberRooms = 0,
                beds = 0,
                price = "",
                src = src,
                sold = false,
                houseTypes = houseTypes.toString(),
                images = listOf(),
                broker = null,
                timestamp = LocalDateTime.now()
            )
        }
    }

    enum class HouseTypes {
        CONDOMINIUM,
        VILLA,
        TOWNHOUSE,
        VACATION_HOME,
        ESTATES_AND_FARMS,
        LAND,
        OTHER_HOUSES
    }
}
